File: composeApp/src/commonMain/kotlin/com/hhp227/concafe/presentation/main/cafemanagement/castlist/CastListUiState.kt
package com.hhp227.concafe.presentation.main.cafemanagement.castlist

import com.hhp227.concafe.domain.model.CafeCastPreview

data class CastListUiState(
    val casts: List<CafeCastPreview> = emptyList(),
    val searchQuery: String = "",
    val deleteTargetCastId: String? = null,
    val nextCursor: String? = null,
    val hasMoreCasts: Boolean = false,
    val isLoadingMore: Boolean = false,
    val isLoading: Boolean = true,
    val infoMessage: String? = null
) {
    val filteredCasts: List<CafeCastPreview>
        get() {
            val query = searchQuery.trim()
            return if (query.isEmpty()) {
                casts
            } else {
                casts.filter { it.name.contains(query, ignoreCase = true) }
            }
        }

    val isSearching: Boolean
        get() = searchQuery.isNotBlank()

    val isDeleteCastDialogVisible: Boolean
        get() = deleteTargetCastId != null
}

File: iosApp/iosApp/Presentation/Main/CafeManagement/CastList/CastListUiState.swift
//
//  CastListUiState.swift
//  ConCafe
//
//  Created by 홍희표 on 2026/07/05.
//

import Foundation
import Shared

struct CastListUiState {
    var casts: [CafeCastPreview] = []
    var searchQuery = ""
    var deleteTargetCastId: String?
    var nextCursor: String?
    var hasMoreCasts = false
    var isLoadingMore = false
    var isLoading = true
    var infoMessage: String?

    var filteredCasts: [CafeCastPreview] {
        let query = searchQuery.trimmingCharacters(in: .whitespacesAndNewlines)

        if query.isEmpty {
            return casts
        } else {
            return casts.filter { $0.name.localizedCaseInsensitiveContains(query) }
        }
    }

    var isSearching: Bool {
        !searchQuery.trimmingCharacters(in: .whitespacesAndNewlines).isEmpty
    }

    var isDeleteCastDialogVisible: Bool {
        deleteTargetCastId != nil
    }
}
